realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -170,14 +174,20 @@ private boolean isAttached() {
      * @throws IndexOutOfBoundsException if {@code location < 0 || location > size()}.
      */
     @Override
-    public void add(int location, E object) {
+    public void add(int location, @Nullable E object) {
```

Comment:
Very, and I agree that we should push for `LinkView` to support it as well, but otherwise the only alternative right now is a complete API split "RealmValueList vs RealmList".